### PR TITLE
chore: Remove allowed_strings in favour of Literal

### PR DESCRIPTION
**Pull request by doublevcodes**
This PR removes the `allowed_strings` converter because we can use `typing.Literal` as a converter to the same effect

### CODE REVIEW

File: bot/exts/info/doc/_cog.py [hunk 1]
```diff
@@ -460,7 +460,7 @@ async def refresh_command(self, ctx: commands.Context) -> None:
     async def clear_cache_command(
         self,
         ctx: commands.Context,
-        package_name: Union[PackageName, allowed_strings("*")]  # noqa: F722
+        package_name: Union[PackageName, Literal["*"]]  # noqa: F722
```
Comment (TizzySaurus) [inline bot/exts/info/doc/_cog.py:463]: The `#noqa` comments are no longer required when using `Literal`.

**Review by TizzySaurus — CHANGES_REQUESTED** [review]

**Review by jchristgit — APPROVED** [review]
Great change, thank you!

Once the review by Tizzy is dealt with, this looks good to merge.

**Comment by MarkKoz** [comment]
I don't think `Literal` does a conversion to lowercase, but I don't think anyone was relying on that feature previously.

File: bot/exts/info/doc/_cog.py (hunk 1 above)
Comment (onerandomusername) [reply bot/exts/info/doc/_cog.py:463]: ^

File: bot/exts/moderation/infraction/management.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ async def infraction_append(
         self,
         ctx: Context,
         infraction: Infraction,
-        duration: t.Union[Expiry, allowed_strings("p", "permanent"), None],   # noqa: F821
+        duration: t.Union[Expiry, t.Literal["p", "permanent"], None],   # noqa: F821
```
Comment (onerandomusername) [inline bot/exts/moderation/infraction/management.py:71]: noqa can be removed

File: bot/exts/moderation/infraction/management.py [hunk 3]
```diff
@@ -107,7 +107,7 @@ async def infraction_edit(
...
```
Comment (onerandomusername) [inline bot/exts/moderation/infraction/management.py:110]: noqa can be removed

**Review by onerandomusername — CHANGES_REQUESTED** [review]
Looks good, just need the noqa removal.

**Comment by ichard26** [comment]
Hey @doublevcodes, it's been a while, do you plan on resuming work on this PR or should we put this up for grabs? Thank you!

**Comment by doublevcodes** [comment]
> Hey @doublevcodes, it's been a while, do you plan on resuming work on this PR or should we put this up for grabs? Thank you!

Hey, sorry for the delay, I can finish this in the coming week.

**Comment by doublevcodes** [comment]
I fixed the noqas, but I did something and now I have 64 changed files 😅

**Comment by Bluenix2** [comment]
Use `git reset` to remove that commit, then rebase your old commit off of `upstream/main` and force push?

**Comment by Xithrius** [comment]
@doublevcodes are you still planning on fixing up the conflicts in this PR?

**Comment by MarkKoz** [comment]
I'm going to finish this up since there has not been a response for over a week. Seems almost done anyway.

**Comment by MarkKoz** [comment]
Unfortunately I used the wrong command and ended up pushing my local main branch to doublevcodes/main, causing the diff to disappear and the PR to close. I can't push to that remote anymore since the PR is closed, and I have no way to reopen the PR. So I will have to just create a new PR unfortunately.

**Pull request closed without merging**